### PR TITLE
Update to 3.0.0-M2

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: docs/tutorial/quotes.md [hunk 1]
```diff
@@ -296,36 +296,36 @@ We also used the `Varargs` extractor to match the arguments of type `Expr[Seq[St
...
+This splice will provide the initial `Quotes` associated with the macro expansion.
 ```scala
-def $[T](x: QuoteContext ?=> Expr[T]): T = ...
+def $[T](x: Quotes ?=> Expr[T]): T = ...
 ```
 
 When we have a splice within a quote, the inner quote context will depend on the outer one.
-This link is represented using the `QuoteContext.Nested` type.
+This link is represented using the `Quotes.Nested` type.
 This relation tells us that it is safe to use expressions created with `q1` within the scope of `q2` but not the other way around (this constraint is statically checked yet).
```
Comment (b-studios) [inline docs/tutorial/quotes.md:325]: What does the `yet` here mean? From the `yet` I would expect a negated form.

Comment (nicolasstucki) [reply docs/tutorial/quotes.md:325]: I need to update this section.

Comment (nicolasstucki) [reply docs/tutorial/quotes.md]: Updated

File: docs/tutorial/quotes.md [hunk 2]
```diff
@@ -296,36 +296,37 @@ We also used the `Varargs` extractor to match the arguments of type `Expr[Seq[St
...
-def $[T](x: QuoteContext ?=> Expr[T]): T = ...
+def $[T](x: Quotes ?=> Expr[T]): T = ...
 ```
 
 When we have a splice within a quote, the inner quote context will depend on the outer one.
-This link is represented using the `QuoteContext.Nested` type.
-This relation tells us that it is safe to use expressions created with `q1` within the scope of `q2` but not the other way around (this constraint is statically checked yet).
+This link is represented using the `Quotes.Nested` type.
+Users of this quotes will almost never need to use `Quotes.Nested`.
+These details are only useful for advanced macros that will inspect code and may encounted quotes and splices.
```
Comment (b-studios) [inline docs/tutorial/quotes.md:326]:
```suggestion
This link is represented using the `Quotes.Nested` type.
Users of quotes will almost never need to use `Quotes.Nested`.
These details are only useful for advanced macros that will inspect code and may encounter details of quotes and splices.
```
I don't understand the last part of line 326 containing "encounted". I changed it to "encounter", is that right?

**Review by b-studios — APPROVED** [review]
Looks good to me, besides this one small comment.

**Pull request merged**